.github/workflows/main.yaml

```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Please change the extension to `yml` to follow the same naming/extension as in the other repos. Moreover, given that this GHA runs the Unite tests, can you please change it to `run-ausf-unit-test.yml`

Reply:
We are running more than unit tests here, we are also building the project and in the future we may want to add more checks (ex. `go vet`).  This workflow file represents the "where/when" rather than the "what" we do.

Reply:
Understood. What about the extension change to "yml" such that we have the same extension for the GHA in the different repos?

Reply:
> We are running more than unit tests here, we are also building the project and in the future we may want to add more checks (ex. `go vet`). This workflow file represents the "where/when" rather than the "what" we do.

BTW, would it be better to have separate GHAs (yml files) for the different things we want to test?

Reply:
I just made the update to use `yml` instead of `yaml`

Reply:
I moved the two validations (build + unit tests) in 2 different jobs but kept it in the same file. My feeling is that it does not need to be in 2 different files at the moment. Having it in 1 file makes it more readable (until the file becomes huge of course but that's not a problem right now).